clients/roscpp/include/ros/service_server_link.h

```diff
@@ -105,7 +105,7 @@ class ROSCPP_DECL ServiceServerLink : public boost::enable_shared_from_this<Serv
    * If there is already a call happening in another thread, this will queue up the call and still block until
    * it has finished.
    */
-  bool call(const SerializedMessage& req, SerializedMessage& resp);
+  bool call(const SerializedMessage& req, SerializedMessage& resp, double timeout = -1);
```

Comment:
This will break ABI compatibility (see http://www.ros.org/reps/rep-0009.html#what-s-not-allowed), but it's very easy to keep ABI compatibility in this case, simply create a new method with the new argument, so you have something like this:

``` c++
bool call(const SerializedMessage& req, SerializedMessage& resp);
bool call(const SerializedMessage& req, SerializedMessage& resp, double timeout);
```

I'm hesitant of providing the new implementation for the 2-argument version here because IIRC if the compiler makes it inline it'd also break ABI compatibility. Therefore, I'd add the implementation on the .cpp.